smarts/core/smarts.py

```diff
@@ -73,12 +73,21 @@ def __init__(
         timestep_sec=0.1,
         reset_agents_only=False,
         zoo_addrs=None,
+        renderer_optional=False,
```

Comment:
If the goal is to make sure that setup is easy even for people that will not use rendering, perhaps even with the option of not installing `panda3d`, we may want this optional to be `True`.

Reply:
Even with disregard to the former point, this still causes issues with basic setup since there is no configuration of the rendering through either of the environments.

Reply:
I agree that we may eventually want to make this default to `True`, although I didn't do it yet because I didn't want to break/change any existing tests or behaviours.  However, I can go ahead and make this change all the way through (including ensuring that the sanity tests all pass without a renderer).  

I will add something for this to the environment interfaces too.

Reply:
Actually, as we just discussed offline, I'm going to try to go either further, only adding a Renderer to Smarts on demand  if there's a sensor that requires it (and getting rid of the `renderer_optional` parameter altogether).  Then I'll try to make the Panda3d dependency optional.